ext/standard/scanf.c

```diff
@@ -618,7 +614,7 @@ PHPAPI int php_sscanf_internal( char *string, char *format,
 	if (numVars) {
 		for (i = varStart;i < argCount;i++){
 			if ( ! Z_ISREF(args[ i ] ) ) {
-				php_error_docref(NULL, E_WARNING, "Parameter %d must be passed by reference", i);
+				ZEND_ASSERT(0 && "Parameter must be passed by reference");
```

Comment:
The whole branch here can be dropped, you just need to leave:
`ZEND_ASSERT(Z_ISREF(args[i]) && "Parameter must be passed by reference");`

Reply:
In which case the constant SCAN_ERROR_VAR_PASSED_BYVAL can be dropped. SCAN_ERROR_INTERNAL is also unused.